### PR TITLE
Replace version numbers with release numbers

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: specification/client_server_api.rst [hunk 1]
```diff
@@ -7,16 +7,6 @@ support both lightweight clients which store no state and lazy-load data from
...
 
-This mostly describes v1 of the Client-Server API as featured in the original September
-2014 launch of Matrix, apart from user-interactive authentication where it is
-encouraged to move to v2, therefore this is the version documented here.
-Version 2 is currently in development (as of Jan-March 2015) as an incremental
-but backwards-incompatible refinement of Version 1 and will be released
-shortly.
-
-Documentation for the old `V1 authentication
-<../attic/v1_registration_login.rst>`_ is still available separately.
```
Comment (richvdh) [inline specification/client_server_api.rst]: should we fix this to point to https://github.com/matrix-org/matrix-doc/blob/master/attic/v1_registration_login.rst?

Comment (illicitonion) [inline specification/client_server_api.rst]: We could do that, but I would be happier pretending it never existed...

**Comment by richvdh** [comment]
What is the plan for exposing old versions of the spec? Presumably they will live somewhere under https://matrix.org/docs/? Should we put the current version there as a sort of r-1?

File: specification/client_server_api.rst [hunk 2]
```diff
@@ -93,41 +83,17 @@ Some requests have unique error codes:
...
-    POST /some/path/here/11?access_token=secret
-    {
-      "key": "This is a post, but it has a txnId."
-    }
-
-    PUT /some/path/here?access_token=secret
-    {
-      "key": "This is a put but it is missing a txnId."
-    }
+Some API endpoints may allow or require the use of POST requests without a
```
Comment (richvdh) [inline specification/client_server_api.rst]: backticks would be nice around POST and PUT for consistency with the previous paragraph

**Comment by illicitonion** [comment]
I was thinking:

https://matrix.org/docs/spec/ serves unstable, but also has links to previous releases, which are served at something like:

File: specification/client_server_api.rst (hunk 2 above)
Comment (illicitonion) [inline specification/client_server_api.rst]: Done

File: specification/application_service_api.rst [hunk 3]
```diff
@@ -153,7 +153,7 @@ application services MUST implement these APIs. These APIs are defined below.
 
 .. _create the user: `sect:asapi-permissions`_
 
-Client-Server v2 API Extensions
+Client-Server API Extensions
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Application services can utilise a more powerful version of the
```
Comment (richvdh) [inline specification/application_service_api.rst]: complete aside but I have a massive pet peeve of people utilising the word "utilise" as a fancy way of saying "use".

Comment (illicitonion) [inline specification/application_service_api.rst]: Done

File: specification/client_server_api.rst [hunk 4]
```diff
@@ -622,15 +581,21 @@ point in time::
...
   
-Clients can add to the stream by POSTing message or state events, and can read
-from the stream via the |initialSync|_, |/rooms/<room_id>/initialSync|_, `Event
-Stream`_ and |/rooms/<room_id>/messages|_ APIs.
+Clients can add to the stream by PUTing message or state events, and can read
+from the stream via the
+|initialSync|_,
+|events|_,
+|/rooms/<room_id>/initialSync|_,
+|/rooms/<room_id>/messages|_,
```
Comment (richvdh) [inline specification/client_server_api.rst]: Needs an 'and' here. With optional Oxford comma.

Comment (richvdh) [inline specification/client_server_api.rst]: maybe on the previous line. ykwim.

Comment (illicitonion) [inline specification/client_server_api.rst]: Done

File: specification/client_server_api.rst [hunk 5]
```diff
@@ -1021,43 +988,41 @@ have to wait in milliseconds before they can try again.
 .. =============================================
 
 .. |createRoom| replace:: ``/createRoom``
-.. _createRoom: /docs/api/client-server/#!/-rooms/create_room
+.. _createRoom: #post-matrix-client-r0-createroom
```
Comment (richvdh) [inline specification/client_server_api.rst]: this looks broken. I'm not sure it's used though?

Comment (illicitonion) [inline specification/client_server_api.rst]: Done

**Comment by richvdh** [comment]
Sounds good. so... should we put the current version there somewhere?

**Comment by illicitonion** [comment]
Done

**Comment by illicitonion** [comment]
In which case I should probably add the links to unstable *after* we cut r0?

**Comment by richvdh** [comment]
I could quibble with the wording, but we've prevaricated long enough here. lgtm, shippit.

**Pull request merged**